clusterctl/clusterdeployer/clusterdeployer_test.go

```diff
@@ -109,24 +125,66 @@ func (c *testClusterClient) GetClusterObjects() ([]*clusterv1.Cluster, error) {
 	return c.clusters, c.GetClusterObjectsErr
 }
 
+func (c *testClusterClient) GetMachineDeploymentObjects() ([]*clusterv1.MachineDeployment, error) {
+	return c.machineDeployments, c.GetMachineDeploymentObjectsErr
+}
+
+func (c *testClusterClient) GetMachineSetObjects() ([]*clusterv1.MachineSet, error) {
+	return c.machineSets, c.GetMachineSetObjectsErr
+}
+
 func (c *testClusterClient) GetMachineObjects() ([]*clusterv1.Machine, error) {
 	return c.machines, c.GetMachineObjectsErr
 }
 
 func (c *testClusterClient) CreateClusterObject(cluster *clusterv1.Cluster) error {
-	if c.CreateClusterObjectErr != nil {
-		return c.CreateClusterObjectErr
+	if c.CreateClusterObjectErr == nil {
```

Comment:
Why not handle the error case first (though admittedly I could be misreading the github diff):

    if c.CreateClusterObjectErr != nil {
        return c.CreateClusterObjectErr
    }
    c.clusters = append(c.clusters, cluster)
    return nil